database/clickhouse/clickhouse.go

```diff
@@ -258,6 +278,17 @@ func (ch *ClickHouse) ensureVersionTable() (err error) {
 	return nil
 }
 
+func (ch *ClickHouse) ensureDistributedTable() error {
+	baseTableName := ch.config.MigrationsTable + "_local"
+	query := fmt.Sprintf(`
+	CREATE TABLE IF NOT EXISTS %s.%s ON CLUSTER %s AS %s.%s Engine=Distributed(%s, %s, %s, sequence) SETTINGS fsync_after_insert = 1`, ch.config.DatabaseName, ch.config.MigrationsTable, ch.config.ClusterName, ch.config.DatabaseName, baseTableName, ch.config.ClusterName, ch.config.DatabaseName, baseTableName)
+
+	if _, err := ch.conn.Exec(query); err != nil {
+		return &database.Error{OrigErr: err, Query: []byte(query)}
+	}
+	return nil
+}
+
 func (ch *ClickHouse) Drop() (err error) {
```

Comment:
do we need to update this for distributed table?

Reply:
hrm, I don't think so because this only drops the migration table not the ch.config.MigrationsTable.
Would be worth testing though.

Reply:
It actually drops both I believe. Though since it just does a `show tables` and loops through deleting everything, we don't need to make any adjustments